cmd/agent/workspace/up.go

```diff
@@ -432,7 +432,7 @@ func CloneRepository(ctx context.Context, sshkey string, local bool, workspaceDi
 
 	// check if command exists
 	if !command.Exists("git") {
-		if local {
+		if workspaceInfo.Agent.Local == "true" {
```

Comment:
how is `workspaceInfo` populated? Do we need to concern ourseleves with nil checking `Agent` or `CLIOptions`

Reply:
Also I assume this may be outside the scope of this PR but why is true a string? Does a user set this? If so maybe we should cast to lowercase during the check in case of "True"

Reply:
`workspaceInfo` is being passed along from the main DevPod binary, we'll error out way earlier if it is `nil`. As for `workspaceInfo.Agent`, it's not a pointer so we should be good. 

The `local` option can both be defined by provider authors in `provider.yaml` and in code which is why it's of type `StrBool`. We could implement a `ToBool` method on it to make it safer to check, wyt?

Reply:
added a `Bool` function to `types.StrBool`